lib/cloud_controller/packager/local_bits_packer.rb

```diff
@@ -4,20 +4,33 @@
 
 module CloudController
   module Packager
+    class ConflictError < StandardError
+    end
+
     class LocalBitsPacker
       include Packager::SharedBitsPacker
 
       def send_package_to_blobstore(blobstore_key, uploaded_package_zip, cached_files_fingerprints)
         tmp_dir = VCAP::CloudController::Config.config.get(:directories, :tmpdir)
-        Dir.mktmpdir('local_bits_packer', tmp_dir) do |root_path|
-          complete_package_path = match_resources_and_validate_package(root_path, uploaded_package_zip, cached_files_fingerprints)
+        local_bits_packer_path = File.join(tmp_dir, "local_bits_packer-#{blobstore_key}")
+
+        begin
+          Dir.mkdir(local_bits_packer_path)
+        rescue StandardError => e
+          raise ConflictError.new("Found a leftover directory that might be from the previous worker’s unfinished job: #{e.message}")
+        end
+
+        begin
+          complete_package_path = match_resources_and_validate_package(local_bits_packer_path, uploaded_package_zip, cached_files_fingerprints)
 
           package_blobstore.cp_to_blobstore(complete_package_path, blobstore_key)
 
           {
             sha1: Digester.new.digest_path(complete_package_path),
             sha256: Digester.new(algorithm: OpenSSL::Digest::SHA256).digest_path(complete_package_path)
           }
+        ensure
+          FileUtils.remove_dir(local_bits_packer_path)
```

Comment:
**suggestion** Curious, what was happening to directories before this change? Also should we have a test for this?

Reply:
I added a test that reproduces the scenario where a folder for a specific GUID already exists on the filesystem.

Reply:
This code mimics the cleanup part of  [Dir.mktmpdir](https://rubyapi.org/o/Dir.mktmpdir#method-c-mktmpdir). I guess we can add something to test this behavior, too.